examples/petsc/ex9p.cpp

```diff
@@ -412,8 +419,8 @@ int main(int argc, char *argv[])
       bool done = false;
       for (int ti = 0; !done; )
       {
-         double dt_real = min(dt, t_final - t);
-         ode_solver->Step(*U, t, dt_real);
+         dt = min(dt, t_final - t);
+         ode_solver->Step(*U, t, dt);
```

Comment:
Was there a specific reason to remove `dt_real`?

Reply:
the `dt` variable is passed by reference to the `Step` method; the PETSc ode solver can use adaptive time stepping, and the `dt` returned by `Step` can be different. Using `dt_real` as it was before was forcing constant time stepping.

Reply:
Hmm, I think there may be some discrepancy between the PETSc ode solver wrapper and the SUNDIALS ode wrappers - the meaning of the `Step` parameters (now documented in the doxygen comment in the base class) may be different. For example, the meaning of `dt` is not to force a time step - at least not in the case when the solver performs time-step control internally - in such cases `dt` should be understood in the sense of a _target_ time-step and the actual time-step should be stored internally.

The above loop was written with that behavior in mind.

I think it is a good idea to try to unify the behavior of the ode solvers. What do you think?

Reply:
I just saw the doxygen documentation. I'll update the PR. Thanks for pointing this out.
BTW: If `Step` can perform multiple internal steps, why not use the `Run` method?

Reply:
Do you mean: make `Run` the primary method and implement `Step` as
`Run(x, t, dt, t + dt);`? There is a use case where this will not work: `Step` is not required to take steps until `t >= t_initial + dt` - it is allowed to take just one step which can be smaller than `dt`. `Run` on the other hand is required to return only after `t >= t_final`. This distinction allows `Step` to support the "one-step" mode of the CVODE and ARKODE time steppers.

Ideally, sub-classes of `ODESolver` will re-implement `Run` to avoid the multiple `Step` calls and the associated overhead.

Reply:
@stefanozampini, just a reminder about this issue... I hope we can merge it soon. -Tzanio